src/phone-number-input/phone-number-input.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useEffect, useRef, useState } from "react";
```

Comment:
Would it be possible to create a wrapper that uses the `InputGroup` instead?

Reply:
I think it's possible but will need to use or enhance the addon list input group. but that introduce confusion as that for more general usage & this phone number specific usage. might need proper discussion on the implementation and proper testing as that might impact existing usage. Could this be part of future improvement?

Reply:
Curious what kind of customisations do you foresee? I thought the existing stuff would have sufficed. 

My idea was to use the `InputGroup` wholesale and for your case, pass the:
- calling codes data as prop of `options`, and customising the `listExtractor` to format the options into the `ListItemDisplayProps` shape for the display
- Then for the value padding, you can just use the `value` prop from the `InputGroup`
- The `clearable` feature can be added onto the `Input` component directly. I think it will definitely benefit other users. Can you also rename this to `allowClear` instead

Let me know if i missed anything or over simplified. We can have a discussion